pandas/tests/io/test_html.py

```diff
@@ -382,10 +382,12 @@ def test_thousands_macau_stats(self, datapath):
 
         assert not any(s.isna().any() for _, s in df.items())
 
-    # https://github.com/pandas-dev/pandas/issues/29622
-    @pytest.mark.xfail(reason='fails for bs4 version >= 4.8.0', strict=False)
     @pytest.mark.slow
     def test_thousands_macau_index_col(self, datapath):
+        # https://github.com/pandas-dev/pandas/issues/29622
+        if self.read_html.keywords.get("flavor") == "bs4":
+            pytest.xfail("fails for bs4 version >= 4.8.0")
+
```

Comment:
did you change this from the decorator in response to my comment?  if so, thats not what i meant.  the decorator is preferred, but without the `strict=False`

Reply:
Yeah just want to xfail one particular parameterization - `bs4` and the parameterization is at class scope - agree decorator is usually preferred.

Updated now - think its easiest to do this in the function itself for this case - unless you know a more elegant way?

Reply:
Just seen I can potentially go for the same approach as you did here https://github.com/pandas-dev/pandas/pull/30521/files?

Reply:
if the test doesnt use the class fixture(s), could it just be moved out of the class?

If not then the approach in 30521 would be my preference, yes.

Reply:
Some class fixtures are used so easier to keep where is. Updated - thanks!